pkg/nmstate/nmstate_test.go

```diff
@@ -1 +1,313 @@
 package nmstate
+
+import (
+	"fmt"
+	"testing"
+
+	nmstatev1 "github.com/nmstate/kubernetes-nmstate/api/v1"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var (
+	v1TestSchemes = []clients.SchemeAttacher{
+		nmstatev1.AddToScheme,
+	}
+	defaultNMStateName = "nmstatename"
+)
+
+func TestPullNMState(t *testing.T) {
+	generateNMState := func(name string) *nmstatev1.NMState {
+		return &nmstatev1.NMState{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Spec: nmstatev1.NMStateSpec{},
+		}
+	}
+
+	testCases := []struct {
+		name                string
+		expectedError       bool
+		addToRuntimeObjects bool
+		expectedErrorText   string
+		client              bool
+	}{
+		{
+			name:                "test1",
+			expectedError:       false,
+			addToRuntimeObjects: true,
+			client:              true,
+		},
+		{
+			name:                "",
+			expectedError:       true,
+			expectedErrorText:   "nmState 'name' cannot be empty",
+			addToRuntimeObjects: true,
+			client:              true,
+		},
+		{
+			name:                "test1",
+			expectedError:       true,
+			expectedErrorText:   "nmState object test1 does not exist",
+			addToRuntimeObjects: false,
+			client:              true,
+		},
+		{
+			name:                "test1",
+			expectedError:       true,
+			expectedErrorText:   "the apiClient cannot be nil",
+			addToRuntimeObjects: true,
+			client:              false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		// Pre-populate the runtime objects
+		var runtimeObjects []runtime.Object
+
+		var testSettings *clients.Settings
+
+		testNmState := generateNMState(testCase.name)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testNmState)
+		}
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects:  runtimeObjects,
+				SchemeAttachers: v1TestSchemes,
+			})
+		}
+
+		// Test the Pull method
+		builderResult, err := PullNMstate(testSettings, testCase.name)
+
+		// Check the error
+		if testCase.expectedError {
+			assert.NotNil(t, err)
+
+			// Check the error message
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, err.Error())
+			}
+		} else {
+			assert.Nil(t, err)
+			assert.Equal(t, testNmState.Name, builderResult.Object.Name)
+			assert.Equal(t, testNmState.Namespace, builderResult.Object.Namespace)
+		}
+	}
+}
+
+func TestNMStateNewBuilder(t *testing.T) {
+	testCases := []struct {
+		name              string
+		expectedErrorText string
+		client            bool
```

Comment:
It doesn't look like the client field is ever used in the test. testSettings is always non-nil on line 126